File: src/shared/ipc/history.ts
import { invoke } from '@tauri-apps/api/core';
import type { HistoryLoadResult, HistoryRecord } from '../types';

export async function loadHistory(): Promise<HistoryLoadResult> {
  return invoke<HistoryLoadResult>('load_history');
}

export async function appendHistory(record: HistoryRecord): Promise<void> {
  return invoke<void>('append_history', { record });
}

export async function clearHistory(): Promise<void> {
  return invoke<void>('clear_history');
}

export async function removeHistory(id: string): Promise<void> {
  return invoke<void>('remove_history', { id });
}

File: src-tauri/src/commands/history.rs
use tauri::State;

use crate::bootstrap::AppState;
use crate::models::HistoryRecord;

#[tauri::command]
pub fn load_history(state: State<'_, AppState>) -> Result<Vec<HistoryRecord>, String> {
    state.history.load_all()
}

#[tauri::command]
pub fn append_history(record: HistoryRecord,
                      state: State<'_, AppState>)
                      -> Result<(), String> {
    state.history.append(&record)
}

#[tauri::command]
pub fn clear_history(state: State<'_, AppState>) -> Result<(), String> {
    state.history.clear()
}

#[tauri::command]
pub fn remove_history(id: String,
                      state: State<'_, AppState>)
                      -> Result<(), String> {
    state.history.remove(&id)
}
